### PR TITLE
[DOC] Improve the explanation on the window parameter (offset) of WindowTransformer

### PR DESCRIPTION
**Pull request by JQGoh**
#1662.

### Summary
 * This improves the documentation such that we explain that the acceptable `window` parameter for an offset is `pandas.Timedelta` data type instead of `pandas.DateOffset`.

### Other Information
 * Also add additional test cases for the class `WindowTransformer` to demonstrate the mentioned behaviors.

### CODE REVIEW

**Comment by JQGoh** [comment]
@madtoinou FYI

File: darts/tests/dataprocessing/transformers/test_window_transformations.py [hunk 1]
```diff
@@ -482,6 +485,46 @@ def test_window_transformer_output(self):
...
+        )
+        self.assertEqual(integer_transformed.components[0], "rolling_mean_4_0")
+
+        invalid_parameters = base_parameters.copy()
+        invalid_parameters.update({"window": pd.DateOffset(hours=4)})
+        invalid_transformer = WindowTransformer(
+            transforms=invalid_parameters,
+        )
+        # if pd.DateOffset, raise ValueError of non-fixed frequency
+        with pytest.raises(ValueError):
```
Comment (madtoinou) [inline darts/tests/dataprocessing/transformers/test_window_transformations.py:525]:
```suggestion
        with self.assertRaises(ValueError):
```

File: darts/tests/dataprocessing/transformers/test_window_transformations.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 import numpy as np
 import pandas as pd
+import pytest
```
Comment (madtoinou) [inline darts/tests/dataprocessing/transformers/test_window_transformations.py:7]: this import is not necessary, see code suggestion below

**Review by madtoinou — COMMENTED** [review]
LGTM 🚀 , the documentation is much clearer with this phrasing!

Minor comments about the snippet checking the `ValueError` raising so that it looks similar to the approach used in the other tests

**Review by madtoinou — APPROVED** [review]
LGTM, thank you for addressing the comments so quickly!

**Review by dennisbader — APPROVED** [review]
Thank you @JQGoh, looks good! 🚀  I'll merge once tests have passed.

**Pull request merged**